### PR TITLE
USWDS - Header: Enable screen reader touch on mobile

### LINKED ISSUE #4787: Mobile navigation dialog makes mobile navigation content inert
### Describe the bug
In #3527, an issue was raised where the mobile navigation dialog was correctly trapping focus, but when using VoiceOver the focus could escape the mobile navigation dialog. This was fixed by adding an `aria-hidden` attribute to everything except the `.usa-header` element (but *only* if that header element is a direct descendant of `<body>`). 

However, this breaks in 2 scenarios, both of which are being demonstrated on https://designsystem.digital.gov. 
1. On https://designsystem.digital.gov itself, the mobile navigation contents, `.usa-nav`, is not a child of the `.usa-header` element, so `.usa-nav` is incorrectly getting an `aria-hidden` attribute. The result is that the focus correctly moves through the mobile navigation dialog when using the TAB key, but it does not when using VoiceOver to view the items in the navigation (e.g. VoiceOver Web Item rotor).
2. In the "Basic header" example on https://designsystem.digital.gov/components/header/ - [direct URL to example](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--default&args=&viewMode=story) - the `.usa-header` element is _not_ a direct descendant of `<body>`, so its parent incorrectly has an `aria-hidden` attribute when the mobile navigation dialog is open.

### Example 1
1. Navigate to https://designsystem.digital.gov/
2. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
3. Tab to and activate the menu button
4. In the developer tools element inspector, note that the `<header class="usa-header">` does not have an `aria-hidden` attribute, but note that the `<nav class="usa-nav site-nav sidenav-mobile">` _does_ have an `aria-hidden` attribute, despite being the content that the user is trying to interact with.

### Example 2
1. Navigate to [the Basic Header example](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--default&args=&viewMode=story) 
2. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
3. Tab to and activate the menu button
4. In the developer tools element inspector, note that the `<header class="usa-header">` element is a child of `<div id="root">`, and that element has an `aria-hidden` attribute.

### Expected behavior

All of the page content has `aria-hidden` applied to it, EXCEPT for the mobile navigation dialog's content.

This can be tested using VoiceOver, by
1. Using ctrl+option+➡️ key combination should cycle through all items in the mobile navigation dialog, but nothing else on the page.
2. Using ctrl+option+U, the VoiceOver Rotor, to view all the mobile navigation dialog's links, but no other links on the page.

This should be the case for a variety of `.usa-header` and `.usa-nav` document structures, since there are likely many different structures in the wild at this point.

### PR DESCRIPTION
**Pull request by amyleadem**
## Summary
This PR makes the mobile header navigation items accessible to users navigating via touch screen readers. 

## Related issue
Closes #4787 

## Preview link
Preview link: [Storybook - Header component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-header-touch/?path=/story/components-header--default)

### 1. Mobile navigation is made inert when nested
When the mobile navigation is activated, all on-page elements (except the navigation and its ancestors) should be made inert by receiving the attributes `aria-hidden="true"` and `data-nav-hidden`. However,  these attributes are currently being applied only to direct children of `body`. This can make the header navigation inert if the page has any structure that nests the header _inside_ a direct child of body. 

The inert mobile navigation is most evident to users who navigate via touch screen readers (ex: VoiceOver Gestures on iOS and TalkBack on Android). For these users, the header nav items are completely invisible, which makes the site unnavigable. 

#### Problem Example
You can see the nesting issue in action in the current [Storybook header component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-header--default). Storybook inserts `div#root` into its page structure as a wrapper for all demo page content. The inert attributes are then applied to `div#root`, which makes all its nested USWDS page content inert, including the header. 

### 2. Mobile menu button is visible to users when menu is open
Additionally, the menu button should be hidden to all users when the menu is open. However, the current state allows the mobile menu button to remain visible to screen readers when the menu is open. 

## Solution
By carefully curating the selector for non-nav elements, we can:
1. Ensure that the header will remain visible to users navigating by touch screen readers
2. Hide the menu button to all users when the menu is open

## Testing and review
To test attributes in markup:
1. Open the [documentation demo page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-header-touch/iframe.html?args=&id=pages-documentation-page--documentation-page&viewMode=story)
1. Narrow the screen to <1025px to active the mobile header structure
1. Inspect the page with web inspector
1. Open the flyout menu by clicking the menu button
1. Confirm that the `aria-hidden="true"`/`data-nav-hidden` attributes:
    1. are NOT applied to `.usa-header`, `.usa-nav-container`,  `.usa-nav`, or the parent of `.usa-header` (like div#root in Storybook)
    1. are applied to the header sibling elements like `usa-section`, `usa-footer`, and `usa-identifier`.

To test behavior in touch screen readers:  
1. Open the documentation page in iOS or android
2. Enable VoiceOver Gestures or TalkBack
3. When menu is open, confirm that:
    1. the menu items can be accessed via both swipe and tap
    2. the menu button can NOT be accessed via swipe or tap

Repeat tests for all header variants.

---

Before opening this PR, make sure you’ve done whichever of these applies to you:

- [ ] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [ ] Run `npm run prettier:scss` to format code.
- [ ] Run `npm test` and confirm that all tests pass.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

### CODE REVIEW

File: packages/usa-nav/src/usa-nav__secondary/usa-nav__secondary.twig [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="usa-nav__secondary {{ modifier }}">
```
Comment (amyleadem) [inline packages/usa-nav/src/usa-nav__secondary/usa-nav__secondary.twig:1]: {{modifier}} is not necessary here

File: packages/usa-header/src/index.js [hunk 2]
```diff
@@ -22,7 +23,7 @@ const CLOSE_BUTTON = `.${PREFIX}-nav__close`;
 const OVERLAY = `.${PREFIX}-overlay`;
 const CLOSERS = `${CLOSE_BUTTON}, .${PREFIX}-overlay`;
 const TOGGLES = [NAV, OVERLAY].join(", ");
-const NON_NAV_ELEMENTS = `body > *:not(${HEADER}):not([aria-hidden])`;
+const NON_NAV_ELEMENTS = `body *:not(${HEADER}, ${NAV_CONTAINER}, ${NAV}, ${NAV} *):not([aria-hidden])`;
```
Comment (amyleadem) [inline packages/usa-header/src/index.js:26]: Make most of the header visible to screen readers while making `.usa-navbar` (and all other page content) invisible.

Hiding `usa-navbar` prevents the user from accessing the page title and menu button while the menu is open.

File: packages/usa-header/src/index.js [hunk 3]
```diff
@@ -43,11 +44,14 @@ const TEMPORARY_PADDING = `${
 }px`;
 
 const hideNonNavItems = () => {
+  const headerParent = document.querySelector(`${HEADER}`).parentNode;
   nonNavElements = document.querySelectorAll(NON_NAV_ELEMENTS);
 
   nonNavElements.forEach((nonNavElement) => {
-    nonNavElement.setAttribute("aria-hidden", true);
-    nonNavElement.setAttribute(NON_NAV_HIDDEN_ATTRIBUTE, "");
+    if(nonNavElement !== headerParent) {
```
Comment (amyleadem) [inline packages/usa-header/src/index.js:51]: Prevents the header parent element from receiving the inert attributes.

**Comment by amyleadem** [comment]
@mejiaj Note: my local unit tests are failing on "navigation toggle - makes all other page content invisible to screen readers" but it passes here. The test template appears to be behaving as expected, so I am a little befuddled. Will you let me know if it is failing your local tests as well?

**Review by mejiaj — APPROVED** [review]
@amyleadem this looks good! 

I didn't see that error on local unit testing. 

```bash
uswds on  al-header-touch [$] via  v16.15.1 took 15s 
➜ npm run test:unit

> @uswds/uswds@3.0.2 test:unit
> gulp unitTests
```

image

**Review by thisisdano — APPROVED** [review]
Good, well-scoped fix

**Pull request merged**